### DISCUSSION TITLE
Ability to define log file location and name

### DISCUSSION THREAD

**Discussion by jagriffin5** [TPC-Council/HammerDB#142] [Q&A] [answered]
I have been searching through the documentation and product and don't see where you have the ability to change the location and name of the log files that are created when the log option is selected.  Is there a way to manipulate this currently and if not, I would strongly suggest an option to add it.  I am trying to build automation around executing baselines and using the log information to provide results but I am finding this challenging due to either permissions issues and/or consistency in file generation depending on what test I am running.

Appreciate any assistance!

**Comment by sm-shaw** [comment, marked as answer]
Setting the location of the log file was already meant to be possible, however a bug that is fixed for v4.0 corrects this. This is defined in issue #125. It can corrected in previous versions by editing  src/generic/genvu.tcl  and adding "global env" just below proc findtempdir as follows: 

```
proc findtempdir {} {
global env
```
Then as per the issue you can define the TMP environment variable and it will pick this up correctly and  use it. There is already the option to use a unique filename but not define one. 
Saying this it is recognised for automation that there needs to be a more straightforward way of picking up results for a particular test run. As a result it is planned to put all of the results in a SQLite database following on from the current web interface version so there is less dependence on a particular logfile location or format. This will not be in the next immediate release - but planned for the one following that.

**Comment by jagriffin5** [comment]
Thank you!  This helped to get me a step closer to my goal.
